Fix various gcc and glibc isues
and more trivial ones.

By Mounir Lamouri (volkmar) for Gentoo

--- src/cybergarage/io/File.cpp.old	2009-06-17 01:53:25.000000000 +0200
+++ src/cybergarage/io/File.cpp	2009-06-17 01:54:06.000000000 +0200
@@ -37,6 +37,7 @@
 #endif
 
 #include <stdio.h>
+#include <string.h>
 
 #if defined(WIN32)
 #include <windows.h>
--- src/cybergarage/io/StringBufferInputStream.cpp.old	2009-06-17 01:58:57.000000000 +0200
+++ src/cybergarage/io/StringBufferInputStream.cpp	2009-06-17 01:59:16.000000000 +0200
@@ -13,6 +13,8 @@
 *
 ******************************************************************/
 
+#include <string.h>
+
 #include <cybergarage/io/StringBufferInputStream.h>
 
 using namespace CyberIO;
--- src/cybergarage/net/DatagramSocket.cpp.old	2009-06-17 02:01:50.000000000 +0200
+++ src/cybergarage/net/DatagramSocket.cpp	2009-06-17 02:02:14.000000000 +0200
@@ -30,6 +30,8 @@
 #include <cybergarage/net/SocketUtil.h>
 
 #include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
 
 using namespace CyberNet;
 
--- src/cybergarage/net/HostInterface.cpp.old	2009-06-17 02:03:40.000000000 +0200
+++ src/cybergarage/net/HostInterface.cpp	2009-06-17 02:03:54.000000000 +0200
@@ -38,6 +38,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #if (defined(WIN32) || defined(__CYGWIN__)) && !defined(ITRON)
 	#include <Iptypes.h>
--- src/cybergarage/net/MulticastSocket.cpp.old	2009-06-17 02:04:19.000000000 +0200
+++ src/cybergarage/net/MulticastSocket.cpp	2009-06-17 02:04:30.000000000 +0200
@@ -28,6 +28,7 @@
 #include <cybergarage/net/HostInterface.h>
 #include <cybergarage/util/StringUtil.h>
 #include <stdio.h>
+#include <string.h>
 
 using namespace CyberNet;
 
--- src/cybergarage/net/SocketUtil.cpp.old	2009-06-17 02:05:47.000000000 +0200
+++ src/cybergarage/net/SocketUtil.cpp	2009-06-17 02:06:23.000000000 +0200
@@ -24,6 +24,7 @@
 #include <cybergarage/net/SocketUtil.h>
 
 #include <stdio.h>
+#include <string.h>
 
 using namespace CyberNet;
 
--- src/cybergarage/http/HTTPHeader.cpp.old	2009-06-17 02:07:57.000000000 +0200
+++ src/cybergarage/http/HTTPHeader.cpp	2009-06-17 02:08:12.000000000 +0200
@@ -17,6 +17,7 @@
 ******************************************************************/
 
 #include <string>
+#include <stdlib.h>
 #include <cybergarage/http/HTTPHeader.h>
 #include <cybergarage/util/StringUtil.h>
 #include <cybergarage/io/StringReader.h>
--- include/cybergarage/http/HTTPPacket.h.old	2009-06-17 13:07:58.000000000 +0200
+++ include/cybergarage/http/HTTPPacket.h	2009-06-17 13:08:17.000000000 +0200
@@ -24,6 +24,7 @@
 #include <string>
 #include <vector>
 #include <sstream>
+#include <stdlib.h>
 
 namespace CyberHTTP {
 
--- src/cybergarage/http/HTTPRequest.cpp.old	2009-06-17 13:08:54.000000000 +0200
+++ src/cybergarage/http/HTTPRequest.cpp	2009-06-17 13:09:09.000000000 +0200
@@ -53,6 +53,7 @@
 #include <string>
 #include <iostream>
 #include <sstream>
+#include <string.h>
 
 using namespace std;
 using namespace CyberNet;
--- src/cybergarage/http/HTTPPacket.cpp.old	2009-06-17 13:10:52.000000000 +0200
+++ src/cybergarage/http/HTTPPacket.cpp	2009-06-17 13:11:04.000000000 +0200
@@ -86,6 +86,7 @@
 #include <string>
 #include <sstream>
 #include <stdlib.h>
+#include <string.h>
 
 using namespace std;
 using namespace CyberHTTP;
--- src/cybergarage/http/HTTPStatus.cpp.old	2009-06-17 13:13:08.000000000 +0200
+++ src/cybergarage/http/HTTPStatus.cpp	2009-06-17 13:13:27.000000000 +0200
@@ -26,6 +26,8 @@
 #include <cybergarage/util/StringTokenizer.h>
 #include <cybergarage/util/StringUtil.h>
 
+#include <stdlib.h>
+
 using namespace CyberHTTP;
 using namespace CyberUtil;
 
--- src/cybergarage/xml/Node.cpp.old	2009-06-17 13:15:02.000000000 +0200
+++ src/cybergarage/xml/Node.cpp	2009-06-17 13:15:34.000000000 +0200
@@ -32,6 +32,8 @@
 #include <cybergarage/xml/Node.h>
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
 using namespace std;
 using namespace CyberXML;
--- include/cybergarage/xml/Parser.h.old	2009-06-17 13:27:05.000000000 +0200
+++ include/cybergarage/xml/Parser.h	2009-06-17 13:27:23.000000000 +0200
@@ -18,6 +18,7 @@
 
 #include <stdio.h>
 #include <string>
+#include <string.h>
 #include <cybergarage/xml/XML.h>
 #include <cybergarage/xml/Node.h>
 #include <cybergarage/io/File.h>
--- src/cybergarage/xml/XML.cpp.old	2009-06-17 13:28:00.000000000 +0200
+++ src/cybergarage/xml/XML.cpp	2009-06-17 13:29:31.000000000 +0200
@@ -51,6 +51,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 
 using namespace std;
 using namespace CyberXML;
@@ -123,7 +124,7 @@
 	char *cpbuf = strdup(str);
 	if (cpbuf == NULL)
 		return NULL;
-	const char *inbuf = cpbuf;
+	char *inbuf = cpbuf;
 	size_t inbyteleft = strlen(str);
 	size_t outbufSize = inbyteleft * sizeof(UnicodeStr) * 4;
 	UnicodeStr *outbuf = new UnicodeStr[outbufSize + 1];
--- src/cybergarage/xml/libxml2/Libxml2Parser.cpp.old	2009-06-17 13:33:14.000000000 +0200
+++ src/cybergarage/xml/libxml2/Libxml2Parser.cpp	2009-06-17 13:33:26.000000000 +0200
@@ -124,7 +124,7 @@
     xmlNodePtr child = cur->xmlChildrenNode;
     while ( child != NULL ) {
 
-      Node *newChildNode = convertToCLinkFormat( doc, child, depth  1 );
+      Node *newChildNode = convertToCLinkFormat( doc, child, depth + 1 );
       if ( newChildNode ) {
 	newNode->addNode( newChildNode );
       }
--- include/cybergarage/upnp/xml/ServiceData.h.old	2009-06-17 13:37:18.000000000 +0200
+++ include/cybergarage/upnp/xml/ServiceData.h	2009-06-17 13:37:24.000000000 +0200
@@ -58,7 +58,7 @@
 
 public:
 
-	void ServiceData::setService(Service *serv)
+	void setService(Service *serv)
 	{
 		service = serv;
 	}
--- include/cybergarage/upnp/Argument.h.old	2009-06-17 13:40:06.000000000 +0200
+++ include/cybergarage/upnp/Argument.h	2009-06-17 13:40:23.000000000 +0200
@@ -20,6 +20,8 @@
 #include <cybergarage/xml/Node.h>
 #include <cybergarage/upnp/xml/ArgumentData.h>
 
+#include <stdlib.h>
+
 namespace CyberLink{
 
 class Service;
--- src/cybergarage/upnp/event/SubscriptionRequest.cpp.old	2009-06-17 13:50:07.000000000 +0200
+++ src/cybergarage/upnp/event/SubscriptionRequest.cpp	2009-06-17 13:50:43.000000000 +0200
@@ -28,6 +28,7 @@
 #include <cybergarage/net/URL.h>
 
 #include <string>
+#include <string.h>
 #include <sstream>
 
 using namespace std;
--- src/cybergarage/upnp/ssdp/SSDP.cpp.old	2009-06-17 14:13:19.000000000 +0200
+++ src/cybergarage/upnp/ssdp/SSDP.cpp	2009-06-17 14:13:31.000000000 +0200
@@ -14,6 +14,8 @@
 ******************************************************************/
 
 #include <string>
+#include <string.h>
+#include <stdlib.h>
 #include <cybergarage/upnp/ssdp/SSDP.h>
 
 using namespace std;
--- src/cybergarage/upnp/media/server/ContentDirectory.cpp.old	2009-06-17 14:16:37.000000000 +0200
+++ src/cybergarage/upnp/media/server/ContentDirectory.cpp	2009-06-17 14:17:11.000000000 +0200
@@ -43,6 +43,7 @@
 #include <cybergarage/upnp/media/server/object/search/IdSearchCap.h>
 
 #include <sstream>
+#include <string.h>
 
 using namespace std;
 using namespace CyberLink;
--- include/cybergarage/upnp/media/server/object/ContentProperty.h.old	2009-06-17 16:04:24.000000000 +0200
+++ include/cybergarage/upnp/media/server/object/ContentProperty.h	2009-06-17 16:04:48.000000000 +0200
@@ -18,6 +18,7 @@
 
 #include <string>
 #include <sstream>
+#include <stdlib.h>
 #include <cybergarage/xml/AttributeList.h>
 
 namespace CyberLink {
--- include/cybergarage/upnp/media/server/object/ContentNode.h.old	2009-06-17 16:05:31.000000000 +0200
+++ include/cybergarage/upnp/media/server/object/ContentNode.h	2009-06-17 16:05:41.000000000 +0200
@@ -19,6 +19,7 @@
 #include <string>
 #include <sstream>
 #include <iostream>
+#include <stdlib.h>
 
 #include <cybergarage/xml/Node.h>
 #include <cybergarage/upnp/media/server/UPnP.h>
--- src/cybergarage/upnp/media/server/object/format/ID3Frame.cpp.old	2009-06-17 16:12:39.000000000 +0200
+++ src/cybergarage/upnp/media/server/object/format/ID3Frame.cpp	2009-06-17 16:12:54.000000000 +0200
@@ -14,6 +14,7 @@
 ******************************************************************/
 
 #include <cybergarage/upnp/media/server/object/format/ID3Frame.h>
+#include <string.h>
 
 using namespace std;
 using namespace CyberLink;
--- src/cybergarage/upnp/media/server/object/format/GIFFormat.cpp.old	2009-06-17 16:17:18.000000000 +0200
+++ src/cybergarage/upnp/media/server/object/format/GIFFormat.cpp	2009-06-17 16:17:29.000000000 +0200
@@ -14,6 +14,7 @@
 ******************************************************************/
 
 #include <stdio.h>
+#include <string.h>
 #include <cybergarage/upnp/media/server/object/format/GIFFormat.h>
 
 using namespace CyberLink;
--- src/cybergarage/upnp/media/server/object/format/PNGFormat.cpp.old	2009-06-17 16:25:40.000000000 +0200
+++ src/cybergarage/upnp/media/server/object/format/PNGFormat.cpp	2009-06-17 16:26:00.000000000 +0200
@@ -14,6 +14,7 @@
 ******************************************************************/
 
 #include <stdio.h>
+#include <string.h>
 #include <cybergarage/upnp/media/server/object/format/PNGFormat.h>
 
 using namespace CyberLink;
--- src/cybergarage/upnp/media/server/object/search/TitleSearchCap.cpp.old	2009-06-17 16:31:57.000000000 +0200
+++ src/cybergarage/upnp/media/server/object/search/TitleSearchCap.cpp	2009-06-17 16:32:06.000000000 +0200
@@ -14,6 +14,7 @@
 ******************************************************************/
 
 #include <string>
+#include <string.h>
 #include <cybergarage/upnp/media/server/object/ContentNode.h>
 #include <cybergarage/upnp/media/server/object/search/TitleSearchCap.h>
 
--- src/cybergarage/upnp/media/server/object/search/IdSearchCap.cpp.old	2009-06-17 16:34:46.000000000 +0200
+++ src/cybergarage/upnp/media/server/object/search/IdSearchCap.cpp	2009-06-17 16:34:54.000000000 +0200
@@ -14,6 +14,7 @@
 ******************************************************************/
 
 #include <string>
+#include <string.h>
 #include <cybergarage/upnp/media/server/object/ContentNode.h>
 #include <cybergarage/upnp/media/server/object/search/IdSearchCap.h>
 
--- sample/clock/ClockDevice.cpp.old	2009-06-17 16:38:22.000000000 +0200
+++ sample/clock/ClockDevice.cpp	2009-06-17 16:38:43.000000000 +0200
@@ -13,6 +13,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <string.h>
 
 using namespace std;
 using namespace CyberLink;
--- sample/tv/TvDevice.cpp.old	2009-06-17 16:42:20.000000000 +0200
+++ sample/tv/TvDevice.cpp	2009-06-17 16:42:51.000000000 +0200
@@ -12,6 +12,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <string.h>
 
 using namespace std;
 using namespace CyberLink;
--- sample/media/server/unix/MediaGateMain.cpp.old	2009-06-17 16:45:39.000000000 +0200
+++ sample/media/server/unix/MediaGateMain.cpp	2009-06-17 17:23:20.000000000 +0200
@@ -20,7 +20,7 @@
 #include <iostream>
 
 #include <stdio.h>
-#include <stdio.h>
+#include <string.h>
 #if defined(WIN32) && !defined(__CYGWIN__)
 #include <conio.h>
 #else
@@ -67,9 +67,9 @@
 	for (n=1; n<argc; n++) {
 #ifdef SUPPORT_MYTHTV
 		if (strcmp(MediaGate::MYTHTV_OPT_STRING, argv[n]) == 0)
-			mode = MediaGate::MYTHTV_MODE;
+			option = MediaGate::MYTHTV_MODE;
 		if (strcmp(MediaGate::MYTHTV_OPT_STRING_OLD, argv[n]) == 0)
-			mode = MediaGate::MYTHTV_MODE;
+			option = MediaGate::MYTHTV_MODE;
 #endif
 		if (strcmp(VERBOSE_ARG, argv[n]) == 0)
 			Debug::on();
